### DISCUSSION TITLE
how to use DIRSTACK

### DISCUSSION THREAD

**Discussion by samyakbardiya** [ajeetdsouza/zoxide#451] [Q&A] [answered]
in the latest release `v0.8.3`, there's a change **Zsh: allow z to navigate dirstack via +n and -n.** so what exactly is this ?

I am guessing this is something like `prevd` and `nextd` of **Fish**.

but I am unable to use them :(

```sh
 ~/video ❯ z +n
zoxide: no match found

 ~/video ❯ zoxide +n
error: Found argument '+n' which wasn't expected, or isn't valid in this context

USAGE:
    zoxide <SUBCOMMAND>

For more information try --help
```

can someone help me out with this one.

**Comment by ajeetdsouza** [comment, marked as answer]
`z` tries to act like a superset of `cd` on most shells. In zsh, you can use the `pushd` and `popd` commands to create a dirstack, which you can then navigate using these commands. For example:

- `cd -3` goes 3 directories back.
- `cd +2` goes 2 directories forward.

Now, you can do the same thing with `z`, i.e. `z -3` and `z +2`. The reason for adding this to zsh is so that people who replace their `cd` command entirely with zoxide (`zoxide init zsh --cmd cd`) don't lose out on any features from the builtin `cd`.

**Reply by samyakbardiya** [reply]
thanks dude, you explained it in detail.

**Reply by martin-braun** [reply]
`pushd` and `popd` aren't equal to `cd`s history. If I continuously call `cd -` I will alternate back and forth, because `cd -` will put my last dir onto the stack again, whereas `popd` will just pop the dir from the stack. I'm looking for the latter, but `z` emulates the former.
